.github/workflows/maven.yml

```diff
@@ -0,0 +1,162 @@
+      # codecov needs to be ran two more times to create codecov reports for the commits made by maven-semantic-release.
+      # See https://github.com/conveyal/gtfs-lib/issues/193. In order to create reports for both the unit and e2e tsts,
+      # the codecov scripts must be ran twice.
+      #
+      # The git commands get the commit hash of the HEAD commit and the commit just before HEAD.
+      - name: Run maven-semantic-release
+        run: |
+          semantic-release --prepare @conveyal/maven-semantic-release --publish @semantic-release/github,@conveyal/maven-semantic-release --verify-conditions @semantic-release/github,@conveyal/maven-semantic-release --verify-release @conveyal/maven-semantic-release --use-conveyal-workflow --dev-branch=dev --skip-maven-deploy
+          if [[ "$GITHUB_REF_SLUG" = "master" ]]; then
+            bash <(curl -s https://codecov.io/bash) -C "$(git rev-parse HEAD)" -s target-unit-test-results -F unit_tests
+            bash <(curl -s https://codecov.io/bash) -C "$(git rev-parse HEAD^)" -s target-unit-test-results -F unit_tests
+            if [ "$SHOULD_RUN_E2E" = "true" ]; then
+              bash <(curl -s https://codecov.io/bash) -C "$(git rev-parse HEAD)" -s target-e2e-test-results -F end_to_end_tests;
+              bash <(curl -s https://codecov.io/bash) -C "$(git rev-parse HEAD^)" -s target-e2e-test-results -F end_to_end_tests;
+            fi
+          fi
+
+
+      - name: Prepare deploy artifacts
+        run: |
+          # get branch name of current branch for use in jar name
+          export BRANCH=$GITHUB_HEAD_REF_SLUG
+          # Replace forward slashes with underscores in branch name.
+          export BRANCH_CLEAN=${BRANCH//\//_}
+          # Create directory that will contain artifacts to deploy to s3.
+          mkdir deploy
+          # Display contents of target directory (for logging purposes only).
+          ls target/*.jar
+          # Copy packaged jar over to deploy dir.
+          cp target/dt-*.jar deploy/
+          # Get the first jar file and copy it into a new file that adds the current branch name. During a
+          # merge to master, there are multiple jar files produced, but they're each effectively the same
+          # code (there may be slight differences in the version shown in the `pom.xml`, but that's not
+          # important for the purposes of creating this "latest branch" jar).
+          ALL_JARS=(target/dt-*.jar)
+          FIRST_JAR="${ALL_JARS[0]}"
+          cp "$FIRST_JAR" "deploy/dt-latest-$BRANCH_CLEAN.jar"
+      - name: Deploy to S3
+        uses: jakejarvis/s3-sync-action@master
+        with:
+          args: --acl public-read
+        env:
+          AWS_S3_BUCKET: datatools-builds
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          SOURCE_DIR: 'deploy'
```

Comment:
This is a security risk to give AWS credentials to a third party. I would have requested that this be changed if I had given a review.